### DISCUSSION TITLE
Possible to swap out markdown renderer?

### DISCUSSION THREAD

**Discussion by faustbrian** [facebook/docusaurus#7408] [Q&A] [answered]
I'm wondering if it is possible to swap out the markdown renderer. For example I would prefer https://markdoc.io/ but stick with Docusaurus for the overall DX it offers out of the box.

Is this possible or is MDX hardcoded and it isn't possible to replace the renderer?

**Comment by slorber** [comment, marked as answer]
I'd like to make this possible to swap the MDX renderer by anything else (including regular MD, Markdoc...), but it might be complicated to achieve technically.

We have built a good amount of tooling on top of the mdx/unified/remark/rehype ecosystem, to handle things like assets and relative.md links, it could be complicated to reproduce all this in a brand new ecosystem without significant effort

I don't think Markdoc can play well with MDX, but willing to see someone do a poc and prove othise. IMHO it's either MDX or Markdoc, not both at the same time

Why would you prefer Markdoc over MDX btw, do you have specific reasons?

**Reply by faustbrian** [reply]
Primarily the syntax for custom components and the ease of extension. Our documentation is written and maintained by non-technical people so MDX is annoying for them at times while they really like Markdoc.

**Reply by slorber** [reply]
> Primarily the syntax for custom components

This is not something we document currently, but in practice, you can swizzle `theme/MDXComponents`, add your own "ambient custom components" there and use them in MDX docs without any ES imports. That's a supported MDX feature.

I believe doing this in MDX:

```mdx
# My custom callout

<callout>Hello</callout>
```

Is not more complicated than the Markdoc equivalent:

```md
# My custom callout

{% callout %}
Hello
{% /callout %}
```

And it will likely render better in GitHub preview too 😉 

> and the ease of extension

What do you mean specifically here? Which kind of extension have you tried adding?

We support remark/rehype plugin which provides a large set of existing community plugins. The ecosystem already exists, unlike Markdoc.

> Our documentation is written and maintained by non-technical people so MDX is annoying for them at times while they really like Markdoc.

Can you give a very specific case where MDX is annoying for them?

How can they like Markdoc, it has just been open-sourced and nobody outside Stripe has practical experience on it yet 😅
